test/box-luatest/space_gc_consumers_test.lua

```diff
@@ -0,0 +1,90 @@
+local t = require('luatest')
+local server = require('luatest.server')
+local cluster = require('luatest.replica_set')
+
+local g = t.group('Upgrade', {{version = '2.11.0'}, {version = '3.0.0'}})
+
+g.before_each(function(cg)
+    cg.server = server:new({
+        datadir = 'test/box-luatest/upgrade/' .. cg.params.version,
+    })
+    cg.server:start()
+end)
+
+g.after_each(function(cg)
+    cg.server:drop()
+end)
+
+g.test_upgrade = function(cg)
+    cg.server:exec(function()
+        local _priv = box.space._priv
+        local space_id = box.schema.GC_CONSUMERS_ID
+
+        t.assert_equals(box.space._space:get(space_id), nil)
+        t.assert_equals(_priv.index.object:select{'space', space_id}, {})
+
+        box.schema.upgrade()
+
+        t.assert_not_equals(box.space._space:get(space_id), nil)
+        t.assert_equals(#_priv.index.object:select{'space', space_id}, 1)
+    end)
+end
+
+g = t.group('Cluster upgrade', {{version = '2.11.0'}, {version = '3.0.0'}})
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new{}
+    local replicaset = {
+        server.build_listen_uri('server1', cg.cluster.id),
+        server.build_listen_uri('server2', cg.cluster.id),
+        server.build_listen_uri('server3', cg.cluster.id)
+    }
+    local box_cfg = {
+        replication_timeout = 0.1,
+        replication = replicaset,
+        election_mode = 'candidate',
+    }
+    cg.servers = {}
+    for i = 1, 3 do
+        local datadir =
+            string.format('test/box-luatest/upgrade/%s-cluster/server%d',
+                          cg.params.version, i)
+        local server = cg.cluster:build_and_add_server{
+            alias = 'server' .. tostring(i),
+            datadir = datadir,
+            box_cfg = box_cfg,
+        }
+        table.insert(cg.servers, server)
+    end
+    cg.cluster:start()
+    cg.cluster:wait_for_fullmesh()
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_cluster_upgrade = function(cg)
+    t.helpers.retrying({}, function(cg)
+        t.assert_not_equals(cg.cluster:get_leader(), nil)
+    end, cg)
+    local leader = cg.cluster:get_leader()
+
+    leader:exec(function()
+        box.schema.upgrade()
+    end)
+
+    -- Wait for all replicas to be upgraded
+    for _, s in pairs(cg.servers) do
+        s:wait_for_vclock_of(leader)
+    end
+
+    -- Check if all gc consumers are present on all replicas
+    for _, s in pairs(cg.servers) do
+        s:exec(function(servers_num)
+            -- No gc consumer for self
+            local consumers_num = servers_num - 1
+            t.assert_equals(#box.space._gc_consumers:select(), consumers_num)
```

Comment:
The test also doesn't work with this commit. Consumers are created only after the next commit

Reply:
Yep, moved this test to the next commit.